homeassistant/components/sensor/pollen.py

```diff
@@ -0,0 +1,349 @@
+
+CONDITIONS = {
+    'allergy_average_forecasted': (
+        'Allergy Index: Forecasted Average',
+        'AllergyAverageSensor',
+        'allergy_average_data',
+        {'data_attr': 'extended_data'},
+        'mdi:flower'
+    ),
+    'allergy_average_historical': (
+        'Allergy Index: Historical Average',
+        'AllergyAverageSensor',
+        'allergy_average_data',
+        {'data_attr': 'historic_data'},
+        'mdi:flower'
+    ),
+    'allergy_index_today': (
+        'Allergy Index: Today',
+        'AllergyIndexSensor',
+        'allergy_index_data',
+        {'key': 'Today'},
+        'mdi:flower'
+    ),
+    'allergy_index_tomorrow': (
+        'Allergy Index: Tomorrow',
+        'AllergyIndexSensor',
+        'allergy_index_data',
+        {'key': 'Tomorrow'},
+        'mdi:flower'
+    ),
+    'allergy_index_yesterday': (
+        'Allergy Index: Yesterday',
+        'AllergyIndexSensor',
+        'allergy_index_data',
+        {'key': 'Yesterday'},
+        'mdi:flower'
+    ),
+    'disease_average_forecasted': (
+        'Cold & Flu: Forecasted Average',
+        'AllergyAverageSensor',
+        'disease_average_data',
+        {'data_attr': 'extended_data'},
+        'mdi:snowflake'
+    )
+}
+
+RATING_MAPPING = [{
+    'label': 'Low',
+    'minimum': 0.0,
+    'maximum': 2.4
+}, {
+    'label': 'Low/Medium',
+    'minimum': 2.5,
+    'maximum': 4.8
+}, {
+    'label': 'Medium',
+    'minimum': 4.9,
+    'maximum': 7.2
+}, {
+    'label': 'Medium/High',
+    'minimum': 7.3,
+    'maximum': 9.6
+}, {
+    'label': 'High',
+    'minimum': 9.7,
+    'maximum': 12
+}]
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ZIP_CODE): cv.positive_int,
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(CONDITIONS)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Configure the platform and add the sensors."""
+    from pypollencom import Client
+
+    _LOGGER.debug('Configuration data: %s', config)
+
+    client = Client(config[CONF_ZIP_CODE])
+    datas = {
+        'allergy_average_data': AllergyAveragesData(client),
+        'allergy_index_data': AllergyIndexData(client),
+        'disease_average_data': DiseaseData(client)
+    }
+
+    for data in datas.values():
+        data.update()
+
+    sensors = []
+    for condition in config.get(CONF_MONITORED_CONDITIONS, []):
+        name, sensor_class, data_key, params, icon = CONDITIONS[condition]
+        sensors.append(globals()[sensor_class](
+            datas[data_key],
+            params,
+            name,
+            icon
+        ))
+
+    add_devices(sensors, True)
+
+
+def average_of_list(list_of_nums, decimal_places=1):
+    """Return the average of a list of ints."""
+    return round(sum(list_of_nums, 0.0)/len(list_of_nums), decimal_places)
+
+
+def calculate_trend(list_of_nums):
+    """Return the average of a list of ints."""
+    ratings = list(
```

Comment:
You can achieve the same with a two level nested generator expression and `next`. That wouldn't be super readable either but I'd prefer that over this.
```py
ratings = next((
    r['label'] for n in list_of_nums
    for r in RATING_MAPPING
    if r['minimum'] <= n <= r['maximum']), None)
```